content/4_Gold/PURS.mdx

```diff
@@ -95,6 +92,105 @@ using std::endl;
 using std::vector;
 
 /** A data structure that can answer point update & range minimum queries. */
+template <class T> class MinSegmentTree {
+  private:
+	const T DEFAULT = std::numeric_limits<T>().max();
+
+	int len;
+	int padded_len;
+	vector<T> segtree;
+
+	T combine(const T &a, const T &b) { return std::min(a, b); }
+
+	void set(int ind, T val, int at, int at_left, int at_right) {
+		if (at_right - at_left == 1) {
```

Comment:
typically for recursive segtree we keep the intervals $$[l, r]$$, not $$[l, r)$$ - altho this is slightly arbitrary

the reason iterative segtree is always $$[l, r)$$ is because its easier to implement that way, whereas for recursive its the same to implement it both ways